### PR TITLE
Max number of students (issue #1135)

### PR DESCRIPTION
**Pull request by akucintavalent**
Resolve issue [#1135](https://github.com/iiuni/projektzapisy/issues/1135) 
Added max_number_of_students field to ThesisFormBase, which resulted in addition this field to ThesisForm and EditThesisForm. If number of selected students is greater than max_number_of_students, the form will throw validation error.

Applied migrations to add this field to the `thesis` table in the database.
Display max_number_of_students on the thesis page.

Created tests for this field. One test with valid form data and one test with not valid form data.
Fixed previous tests.

### CODE REVIEW

**Review by mbuszka — CHANGES_REQUESTED** [review]

File: zapisy/apps/theses/templates/theses/thesis.html [hunk 1]
```diff
@@ -81,6 +81,15 @@ <h1 class="d-inline-block">
                 {% endif %}
             </td>
         </tr>
+        <tr>
+          <th scope="row">Maskymalna liczba studentów</th>
```
Comment (JMendyk) [inline zapisy/apps/theses/templates/theses/thesis.html:85]:
```suggestion
          <th scope="row">Maksymalna liczba studentów</th>
```

File: zapisy/apps/theses/forms.py [hunk 2]
```diff
@@ -44,15 +45,19 @@ class Meta:
         queryset=Student.objects.all(),
         required=False,
         label="Przypisani studenci",
-        help_text=(f"Limit przypisanych studentów: {MAX_ASSIGNED_STUDENTS}. "
-                   "Aby przypisać więcej studentów należy zwrócić się do zastępcy dyrektora ds. dydaktycznych."),
+        # help_text=(f"Limit przypisanych studentów: {MAX_ASSIGNED_STUDENTS}. "
+        #     "Aby przypisać więcej studentów należy zwrócić się do zastępcy"
+        #     " dyrektora ds. dydaktycznych."),
```
[the remark this answers is not in the document]
Comment (akucintavalent) [reply zapisy/apps/theses/forms.py:50]: Zrobione

File: zapisy/apps/theses/forms.py [hunk 3]
```diff
@@ -73,11 +78,11 @@ def __init__(self, user, *args, **kwargs):
         self.helper = FormHelper()
         self.helper.form_method = 'POST'
 
-    def clean_students(self):
+    def clean(self):
```
[the remark this answers is not in the document]
Comment (akucintavalent) [reply zapisy/apps/theses/forms.py:81]: Zrobione

File: zapisy/apps/theses/forms.py [hunk 4]
```diff
@@ -73,11 +78,11 @@ def __init__(self, user, *args, **kwargs):
...
         students = self.cleaned_data['students']
-        if 'students' in self.changed_data and len(students) > MAX_ASSIGNED_STUDENTS:
+        max_number_of_students = int(self.cleaned_data['max_number_of_students'])
+        if 'students' in self.changed_data and len(students) > max_number_of_students:
```
[the remark this answers is not in the document]
Comment (akucintavalent) [reply zapisy/apps/theses/forms.py:84]: Na razie nie rozumiem jak ta funkcja działa

**Comment by akucintavalent** [comment]
@mbuszka, could you please review the code again?

**Review by mbuszka — APPROVED** [review]

**Pull request merged**